analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ClearTextProtocolsAreSensitive.cs

```diff
@@ -61,16 +55,54 @@ public sealed class ClearTextProtocolsAreSensitive : HotspotDiagnosticAnalyzer
             {"clear-text SMTP", "SMTP over SSL/TLS or SMTP with STARTTLS" }
         };
 
+        private readonly List<string> commonlyUsedXmlDomains = new List<string>
+        {
+            "www.w3.org",
+            "xml.apache.org",
+            "schemas.xmlsoap.org",
+            "schemas.openxmlformats.org",
+            "rdfs.org",
+            "purl.org",
+            "xmlns.com",
+            "schemas.google.com",
+            "a9.com",
+            "ns.adobe.com",
+            "ltsc.ieee.org",
+            "docbook.org",
+            "graphml.graphdrawing.org",
+            "json-schema.org"
+        };
+
+        private readonly List<string> commonlyUsedExampleDomains = new List<string> {"example.com", "example.org", "test.com"};
+        private readonly List<string> localhostAddresses = new List<string> {"localhost", "127.0.0.1", "::1"};
+
         private readonly CSharpObjectInitializationTracker objectInitializationTracker =
             new CSharpObjectInitializationTracker(constantValue => constantValue is bool value && value,
                                                   ImmutableArray.Create(KnownType.System_Net_Mail_SmtpClient, KnownType.System_Net_FtpWebRequest),
                                                   propertyName => propertyName == EnableSslName);
 
+        private readonly Regex httpRegex;
+        private readonly Regex ftpRegex;
+        private readonly Regex telnetRegex;
+        private readonly Regex telnetRegexForIdentifier;
+        private readonly Regex validServerRegex;
+
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, EnableSslRule);
 
         public ClearTextProtocolsAreSensitive() : this(AnalyzerConfiguration.Hotspot) { }
 
-        public ClearTextProtocolsAreSensitive(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration) { }
+        public ClearTextProtocolsAreSensitive(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration)
+        {
+            const string allSubdomainsPattern = "([^/]+\\.)?";
+            var domainsList = commonlyUsedXmlDomains.Concat(commonlyUsedExampleDomains.Select(x => allSubdomainsPattern + x)).Concat(localhostAddresses);
```

Comment:
I'd swap the order to make it easier to read (simple stuff together, more complicated at the end)
```suggestion
            var domainsList = commonlyUsedXmlDomains.Concat(localhostAddresses).Concat(commonlyUsedExampleDomains.Select(x => allSubdomainsPattern + x));
```

Reply:
Update: This is even better. localhost, then common domains, than example domains (to star simple and keep domains together)
```suggestion
            var domainsList = localhostAddresses.Concat(commonlyUsedXmlDomains).Concat(commonlyUsedExampleDomains.Select(x => allSubdomainsPattern + x));
```